### PR TITLE
fix: Enable errcheck check-type-assertions in golangci

### LINKED ISSUE #151: Enable errcheck check-type-assertions in golangci
```
linters-settings:
  errcheck:
    check-type-assertions: false
    check-blank: false
```

### PR DESCRIPTION
**Pull request by fqutishat**
closes #151

Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

Enable check-type-assertions in golangci

### CODE REVIEW

File: pkg/didmethod/peer/did.go [hunk 1]
```diff
@@ -65,7 +65,10 @@ func validateDID(doc *did.Doc) error {
 
 	peerDid := doc.ID
 
-	matched, _ := regexp.MatchString(`did:peer:11-([a-fA-F0-9]){64}`, peerDid)
+	matched, err := regexp.MatchString(`did:peer:11-([a-fA-F0-9]){64}`, peerDid)
+	if err != nil {
+		return err
```
Comment (troyronda) [inline pkg/didmethod/peer/did.go]: this should probably be wrapped by xerrors?

File: pkg/didmethod/httpbinding/resolver.go [hunk 2]
```diff
@@ -108,7 +108,10 @@ func New(endpointURL string, opts ...ResolverOpt) (*DIDResolver, error) {
 
 // Read implements didresolver.DidMethod.Read interface (https://w3c-ccg.github.io/did-resolution/#resolving-input)
 func (res *DIDResolver) Read(DID string, _ ...didresolver.ResolveOpt) ([]byte, error) {
-	reqURL, _ := url.ParseRequestURI(res.endpointURL)
+	reqURL, err := url.ParseRequestURI(res.endpointURL)
+	if err != nil {
+		return nil, err
```
Comment (troyronda) [inline pkg/didmethod/httpbinding/resolver.go]: this should probably be wrapped by xerrors?

Comment (fqutishat) [reply pkg/didmethod/httpbinding/resolver.go]: done

File: pkg/didmethod/peer/did.go (hunk 1 above)
Comment (fqutishat) [reply pkg/didmethod/peer/did.go]: done

File: pkg/didmethod/peer/did.go [hunk 3]
```diff
@@ -65,7 +65,10 @@ func validateDID(doc *did.Doc) error {
 
 	peerDid := doc.ID
 
-	matched, _ := regexp.MatchString(`did:peer:11-([a-fA-F0-9]){64}`, peerDid)
+	matched, err := regexp.MatchString(`did:peer:11-([a-fA-F0-9]){64}`, peerDid)
```
Comment (troyronda) [inline pkg/didmethod/peer/did.go]: TODO: regex should have been compiled

**Pull request merged**